readthedocs/doc_builder/environments.py

```diff
@@ -521,8 +525,15 @@ def __enter__(self):
                                       .format(self.container_id))))
                     client = self.get_client()
                     client.remove_container(self.container_id)
-        except DockerAPIError:
+        except (DockerAPIError, ConnectionError):
+            # If there is an exception here, we swallow the exception as this
+            # was just during a sanity check anyways.
             pass
+        except BuildEnvironmentError:
+            # There may have been a problem connecting to Docker altogether, or
+            # some other handled exception here.
+            self.__exit__(*sys.exc_info())
```

Comment:
Will this not automatically call `__exit__`?

Reply:
It won't, because this is raised in `__enter__`. Errors raised in `__enter__` raise to the parent context, so we have to explicitly clean up here.